File: backend/app/Models/Utilisateur.php
<?php

namespace App\Models;

use Illuminate\Foundation\Auth\User as Authenticatable;
use Illuminate\Notifications\Notifiable;
use Laravel\Sanctum\HasApiTokens;

class Utilisateur extends Authenticatable
{
    use HasApiTokens, Notifiable;

    protected $table = 'utilisateurs';

    protected $fillable = [
        'nom',
        'prenom',
        'email',
        'telephone',
        'adresse',
        'mot_de_passe',
        'photo',
        'role',
        'statut',
    ];

    protected $hidden = [
        'mot_de_passe',
    ];

    protected function casts(): array
    {
        return [
            'mot_de_passe' => 'hashed',
        ];
    }

    // Laravel utilise "password" par défaut — on redirige vers mot_de_passe
    public function getAuthPasswordName(): string
    {
        return 'mot_de_passe';
    }

    // Relations vers les profils spécifiques
    public function producteur()
    {
        return $this->hasOne(Producteur::class, 'utilisateur_id');
    }

    public function entrepot()
    {
        return $this->hasOne(Entrepot::class, 'utilisateur_id');
    }

    public function acheteurGros()
    {
        return $this->hasOne(AcheteurGros::class, 'utilisateur_id');
    }

    public function transporteur()
    {
        return $this->hasOne(Transporteur::class, 'utilisateur_id');
    }
}

File: mobile-app/lib/models/utilisateur.dart
class Utilisateur {
  final int id;
  final String nom;
  final String prenom;
  final String email;
  final String? telephone;
  final String role;
  final String statut;
  final Map<String, dynamic>? producteur;
  final Map<String, dynamic>? entrepot;
  final Map<String, dynamic>? acheteurGros;
  final Map<String, dynamic>? transporteur;

  const Utilisateur({
    required this.id,
    required this.nom,
    required this.prenom,
    required this.email,
    this.telephone,
    required this.role,
    required this.statut,
    this.producteur,
    this.entrepot,
    this.acheteurGros,
    this.transporteur,
  });

  factory Utilisateur.fromJson(Map<String, dynamic> j) => Utilisateur(
        id: j['id'],
        nom: j['nom'],
        prenom: j['prenom'],
        email: j['email'],
        telephone: j['telephone'],
        role: j['role'],
        statut: j['statut'],
        producteur: j['producteur'] as Map<String, dynamic>?,
        entrepot: j['entrepot'] as Map<String, dynamic>?,
        // Laravel sérialise la relation Eloquent acheteurGros() en snake_case.
        acheteurGros: j['acheteur_gros'] as Map<String, dynamic>?,
        transporteur: j['transporteur'] as Map<String, dynamic>?,
      );

  String get nomComplet => '$prenom $nom';
}
